cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/CassandraDialect.java

```diff
@@ -0,0 +1,329 @@
+			throws TupleAlreadyExistsException {
+
+		List<TupleOperation> updateOps = new ArrayList<TupleOperation>( tuple.getOperations().size() );
+		List<TupleOperation> deleteOps = new ArrayList<TupleOperation>( tuple.getOperations().size() );
+
+		for ( TupleOperation op : tuple.getOperations() ) {
+			switch ( op.getType() ) {
+				case PUT:
+					updateOps.add( op );
+					break;
+				case REMOVE:
+				case PUT_NULL:
+					deleteOps.add( op );
+					break;
+				default:
+					throw new HibernateException( "TupleOperation not supported: " + op.getType() );
+			}
+		}
+
+		if ( deleteOps.size() > 0 ) {
+
+			Delete.Selection deleteSelection = delete();
+			for ( TupleOperation tupleOperation : deleteOps ) {
+				deleteSelection.column( quote( tupleOperation.getColumn() ) );
+			}
+			Delete delete = deleteSelection.from( quote( key.getTable() ) );
+			Delete.Where deleteWhere = delete.where(
+					eq(
+							quote( key.getColumnNames()[0] ),
+							QueryBuilder.bindMarker()
+					)
+			);
+			for ( int i = 1; i < key.getColumnNames().length; i++ ) {
+				deleteWhere = deleteWhere.and( eq( quote( key.getColumnNames()[i] ), QueryBuilder.bindMarker() ) );
+			}
+
+			bindAndExecute( key.getColumnValues(), delete );
+		}
+
+		if ( updateOps.size() > 0 ) {
+
+			Insert insert = insertInto( quote( key.getTable() ) );
+			Object[] columnValues = new Object[updateOps.size()];
+			for ( int i = 0; i < updateOps.size(); i++ ) {
+				TupleOperation op = updateOps.get( i );
+				insert.value( quote( op.getColumn() ), QueryBuilder.bindMarker() );
+				columnValues[i] = op.getValue();
+			}
+
+			bindAndExecute( columnValues, insert );
+		}
+	}
+
+	@Override
+	public void removeTuple(EntityKey key, TupleContext tupleContext) {
+
+		Delete delete = delete().from( quote( key.getTable() ) );
+		Delete.Where deleteWhere = delete.where(
+				eq( quote( key.getColumnNames()[0] ), QueryBuilder.bindMarker() )
+		);
+		for ( int i = 1; i < key.getColumnNames().length; i++ ) {
+			deleteWhere = deleteWhere.and( eq( quote( key.getColumnNames()[i] ), QueryBuilder.bindMarker() ) );
+		}
+
+		bindAndExecute( key.getColumnValues(), delete );
+	}
+
+	@Override
+	public Association getAssociation(AssociationKey key, AssociationContext associationContext) {
+		return null; // TODO
+	}
+
+	@Override
+	public Association createAssociation(AssociationKey key, AssociationContext associationContext) {
+		return null; // TODO
+	}
+
+	@Override
+	public void insertOrUpdateAssociation(
+			AssociationKey key,
+			Association association,
+			AssociationContext associationContext) {
+
+		// TODO
+	}
+
+	@Override
+	public void removeAssociation(AssociationKey key, AssociationContext associationContext) {
+		// TODO
+	}
+
+	@Override
+	public GridType overrideType(Type type) {
+		return CassandraTypeMapper.INSTANCE.overrideType( type );
+	}
+
+	@Override
+	public void executeBatch(OperationsQueue queue) {
+
+		// placeholder until we get around to using logged batch for these.
```

Comment:
You can skip the method altogether by implementing `GridDialect` rather than `BatchableGridDialect`.